projects/igniteui-angular/src/lib/grids/grid-navigation.service.ts

```diff
@@ -208,35 +216,37 @@ export class IgxGridNavigationService {
 
     public navigateTop(visibleColumnIndex) {
         const verticalScroll = this.grid.verticalScrollContainer.getVerticalScroll();
+        const cellSelector = this.isTreeGrid && visibleColumnIndex === 0 ? 'igx-tree-grid-cell' : 'igx-grid-cell';
         if (verticalScroll.scrollTop === 0) {
             const cells = this.grid.nativeElement.querySelectorAll(
-                `igx-grid-cell[data-visibleIndex="${visibleColumnIndex}"]`);
+                `${cellSelector}[data-visibleIndex="${visibleColumnIndex}"]`);
             cells[0].focus();
         } else {
             this.grid.nativeElement.focus({preventScroll: true});
             this.grid.verticalScrollContainer.scrollTo(0);
             this.grid.verticalScrollContainer.onChunkLoad
                 .pipe(first()).subscribe(() => {
                     const cells = this.grid.nativeElement.querySelectorAll(
-                        `igx-grid-cell[data-visibleIndex="${visibleColumnIndex}"]`);
-                if (cells.length > 0) { cells[0].focus(); }
+                        `${cellSelector}[data-visibleIndex="${visibleColumnIndex}"]`);
+                    if (cells.length > 0) { cells[0].focus(); }
                 });
         }
     }
 
     public navigateBottom(visibleColumnIndex) {
         const verticalScroll = this.grid.verticalScrollContainer.getVerticalScroll();
+        const cellSelector = this.isTreeGrid && visibleColumnIndex === 0 ? 'igx-tree-grid-cell' : 'igx-grid-cell';
         if (verticalScroll.scrollTop === verticalScroll.scrollHeight - this.grid.verticalScrollContainer.igxForContainerSize) {
             const cells = this.grid.nativeElement.querySelectorAll(
-                `igx-grid-cell[data-visibleIndex="${visibleColumnIndex}"]`);
+                `${cellSelector}[data-visibleIndex="${visibleColumnIndex}"]`);
             cells[cells.length - 1].focus();
```

Comment:
Similar to the previous comment - Is there any chance to have no cells?

Reply:
This is a case when press ArrowDown+Ctrl keys on a cell and do not need to scroll. There is no change cells to be empty.